src/libraries/SettlementGuard.sol

```diff
@@ -24,7 +24,6 @@ library SettlementGuard {
         address currentLocker = getLocker();
 
         // either set from non-zero to zero (set) or from zero to non-zero (reset)
-        if (currentLocker == newLocker) return;
         if (currentLocker != address(0) && newLocker != address(0)) revert IVault.LockerAlreadySet(currentLocker);
 
         assembly ("memory-safe") {
```

Comment:
I think we should change the value , from address to bool.
If locked , no one can re-entry.
what do you think ?

Reply:
So this will be like the ReentrancyGuard (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/ReentrancyGuard.sol) 

The lock function will have the nonReentrant modifier, while the other isLock functions will check that _status is ENTERED. 

If so, it seems like we can't use 1 (NOT_ENTERED) and 2 (ENTERED) like the OZ one since using tstore/tload, the data will be reset to 0. 

https://soliditylang.org/blog/2024/01/26/transient-storage/

> The new data location behaves as a key-value store similar to storage with the main difference being that data in transient storage is not permanent, but is scoped to the current transaction only, after which it will be reset to zero. 

So 0 and 1, or bool (false and true), both are the same functionally, but you might want to test if there is any gas cost difference.

Reply:
personally think this PR does resolve zellic-r1. 

though agree with the pointer above, created https://github.com/pancakeswap/pancake-v4-core/issues/123 to track which is more of a gas optimisation potential